### PR TITLE
Support cipher chacha20-poly1305@openssh.com

### LINKED ISSUE #629: Adding support for chacha20-poly1305@openssh.com
I suggest you add support for chacha20-poly1305@openssh.com cipher. It is one of the default ciphers in OpenSSH [https://man.openbsd.org/ssh_config.5#Ciphers](https://man.openbsd.org/ssh_config.5#Ciphers)

### PR DESCRIPTION
**Pull request by hpoettker**
Resolves #629.

### CODE REVIEW

File: src/main/java/com/hierynomus/sshj/transport/cipher/ChachaPolyCipher.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+            mac = null;
+            throw new SSHRuntimeException(e);
+        }
+
+        setSequenceNumber(0);
+    }
+
+    @Override
+    public void setSequenceNumber(long seq) {
+        byte[] seqAsBytes = LONG_TO_BYTES_BUFFER.putLong(0, seq).array();
```
Comment (vladimirlagunov) [inline src/main/java/com/hierynomus/sshj/transport/cipher/ChachaPolyCipher.java:82]: `LONG_TO_BYTES_BUFFER` is static. If there are two concurrent `setSequenceNumber` from different instances are called simultaneously, one of them can get wrong results.

**Review by vladimirlagunov — COMMENTED** [review]
It's a very important feature! Thanks for implementing it!

File: src/main/java/com/hierynomus/sshj/transport/cipher/ChachaPolyCipher.java (hunk 1 above)
Comment (hierynomus) [reply src/main/java/com/hierynomus/sshj/transport/cipher/ChachaPolyCipher.java:82]: Haha, you beat me to it ;) Indeed, there is a big concurrency issue here. Prevent static mutable fields such as buffers etc.

**Review by hierynomus — APPROVED** [review]

**Comment by hierynomus** [comment]
Thanks for the PR, I fixed the comment @vladimirlagunov had, and merged it.

**Comment by dkocher** [comment]
Wouldn't want you add this to the default configuration in `DefaultConfig`?

**Comment by hierynomus** [comment]
Thanks @dkocher, indeed, fixed on master!

**Pull request merged**